pkg/controller/manage/manage_controller.go

```diff
@@ -203,6 +127,176 @@ func NewKmeshManageController(client kubernetes.Interface, security *kmeshsecuri
+	if utils.ShouldEnroll(nil, NS) {
+		log.Infof("enable Kmesh for all pods in namespace: %s", NS.Name)
+		enableKmeshForPodsInNamespace(NS.Name, podLister, queue, security)
+	}
+}
+
+func handleNamespaceUpdateFunc(oldObj, newObj interface{}, podLister v1.PodLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	oldNS, okOld := oldObj.(*corev1.Namespace)
+	newNS, okNew := newObj.(*corev1.Namespace)
+	if !okOld || !okNew {
+		log.Errorf("Expected *corev1.Namespace but got %T and %T", oldObj, newObj)
+		return
+	}
+
+	// Compare labels to check if they have actually changed
+	if !utils.ShouldEnroll(nil, oldNS) && utils.ShouldEnroll(nil, newNS) {
+		log.Infof("Enabling Kmesh for all pods in namespace: %s", newNS.Name)
+		enableKmeshForPodsInNamespace(newNS.Name, podLister, queue, security)
+	}
+
+	if utils.ShouldEnroll(nil, oldNS) && !utils.ShouldEnroll(nil, newNS) {
+		log.Infof("Disabling Kmesh for all pods in namespace: %s", newNS.Name)
+		disableKmeshForPodsInNamespace(newNS.Name, podLister, queue, security)
+	}
+}
+
+func enableKmeshManage(pod *corev1.Pod, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	sendCertRequest(security, pod, kmeshsecurity.ADD)
+	if !isPodReady(pod) {
+		log.Debugf("Pod add event: %s/%s is not ready, skipping Kmesh manage enable", pod.GetNamespace(), pod.GetName())
+		return
+	}
+	log.Infof("%s/%s: enable Kmesh manage", pod.GetNamespace(), pod.GetName())
+	nspath, _ := ns.GetPodNSpath(pod)
+	if err := utils.HandleKmeshManage(nspath, true); err != nil {
+		log.Errorf("failed to enable Kmesh manage")
+		return
+	}
+	queue.AddRateLimited(QueueItem{podName: pod.Name, podNs: pod.Namespace, action: ActionAddAnnotation})
+}
+
+func disableKmeshManage(pod *corev1.Pod, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	sendCertRequest(security, pod, kmeshsecurity.DELETE)
+	if !isPodReady(pod) {
+		log.Debugf("Pod add event: %s/%s is not ready, skipping Kmesh manage enable", pod.GetNamespace(), pod.GetName())
+		return
+	}
+	log.Infof("%s/%s: disable Kmesh manage", pod.GetNamespace(), pod.GetName())
+	nspath, _ := ns.GetPodNSpath(pod)
+	if err := utils.HandleKmeshManage(nspath, false); err != nil {
+		log.Errorf("failed to disable Kmesh manage")
+		return
+	}
+	queue.AddRateLimited(QueueItem{podName: pod.Name, podNs: pod.Namespace, action: ActionDeleteAnnotation})
+}
+
+func enableKmeshForPodsInNamespace(namespace string, podLister v1.PodLister, queue workqueue.RateLimitingInterface, security *kmeshsecurity.SecretManager) {
+	pods, err := podLister.Pods(namespace).List(labels.Everything())
+	if err != nil {
+		log.Errorf("Error listing pods: %v", err)
+		return
+	}
+
+	if len(pods) == 0 {
+		clientset, err := utils.GetK8sclient()
+		if err != nil {
+			log.Errorf("Error getting Kubernetes client: %v", err)
+			return
+		}
+
+		podList, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
```

Comment:
We donot need this, if the pod lister has no pod, it means no pod event has received. We can rely on pod informer to trigger manage later,